src/health-check/main.cpp

```diff
@@ -132,16 +132,14 @@ class HealthCheckerProcess : public ProtobufProcess<HealthCheckerProcess>
   {
     VLOG(1) << "Check passed";
 
-    // Send a healthy status update on the first success,
-    // and on the first success following failure(s).
-    if (initializing || consecutiveFailures > 0) {
-      TaskHealthStatus taskHealthStatus;
```

Comment:
+1 to @drexin's comment; cutting down communication between the task host and the scheduler was the original motivation behind implementing health checks in the Mesos agent :)

Other options: 1) let the framework persist task health status before acknowledging the update 2) have a deeper look at the status updates that are synthesized by the master during reconciliation, maybe trigger the executor to send the current health?